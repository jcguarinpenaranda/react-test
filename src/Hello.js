import React from 'react'

class Hello extends React.Component {
  constructor(props) {
    super(props);
    
    // el valor debe iniciar en 0
    this.state = {
      num: 0,
    
    }
    
    this.sumarNumero = this.sumarNumero.bind(this);
    this.restarNumero = this.restarNumero.bind(this);
    this.doblarNumero = this.doblarNumero.bind(this);
    this.dividirNumero = this.dividirNumero.bind(this);
  }
  
  // al ejecutarse, el numero cambia +1
  sumarNumero() {
    const numActual = this.state.num;
    this.setState({ num: numActual + 1 });
  }
  
  // al ejecutarse, el numero cambia -1
  restarNumero() {
    const numActual = this.state.num;
    this.setState({ num: numActual - 1 });
  }
   doblarNumero() {
    const numActual = this.state.num;
    this.setState({ num: numActual * 2});
  }
    
   dividirNumero() {
    const numActual = this.state.num;
    this.setState({ num: numActual / 2});
  }
  
  render() {
    return (
      <div>
        <p>{this.state.num}</p>
        {/* probar los t√≠tulos de los botones */}
        <button onClick={this.sumarNumero}>Sumar</button>
        <button onClick={this.restarNumero}>Restar</button>
        <button onClick={this.doblarNumero}>Duplicar</button>
         <button onClick={this.dividirNumero}>Dividir</button>
      </div>
    );
  }
}

export default Hello