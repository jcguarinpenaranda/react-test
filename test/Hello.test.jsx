import React from 'react';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';

import Hello from '../src/Hello';

test('should be defined', () => {
  expect(Hello).toBeTruthy();
});

test('should mount correctly', () => {
  const m = mount(<Hello />);
  expect(m.html()).toBeTruthy();
});

describe('initial state', () => {
  const m = mount(<Hello />);

  test('num should be 0', () => {
    expect(m.instance().state.num).toEqual(0);
  });
});

describe('#sumarNumero', () => {
  test('should be ok', () => {
    const m = mount(<Hello />);
    m.instance().sumarNumero();
    expect(m.instance().state.num).toEqual(1);
  });
});

describe('#restarNumero', () => {
  test('should be ok', () => {
    const m = mount(<Hello />);
    m.instance().setState({ num: 90 });
    m.instance().restarNumero();
    expect(m.instance().state.num).toEqual(89);
  });
});

describe('#doblarNumero', () => {
  test('should be ok', () => {
    const m = mount(<Hello />);
    m.instance().setState({ num: 5 });
    m.instance().doblarNumero();
    expect(m.instance().state.num).toEqual(10);
  });
});

describe('#dividirNumero', () => {
  test('should be ok', () => {
    const m = mount(<Hello />);
    m.instance().setState({ num: 5 });
    m.instance().dividirNumero();
    expect(m.instance().state.num).toEqual(2.5);
  });
});

describe('Rendered view', () => {
  test('add button should say "Sumar"', () => {
    const m = mount(<Hello />);
    expect(m.find('.buttonSumar').text()).toEqual('Sumar');
  });

  test('should match snapshot', () => {
    const m = mount(<Hello />);
    expect(toJSON(m)).toMatchSnapshot();
  });
});
