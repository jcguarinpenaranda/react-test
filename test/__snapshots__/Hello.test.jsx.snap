// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendered view should match snapshot 1`] = `
<Hello>
  <div>
    <p>
      0
    </p>
    <button
      className="buttonSumar"
      onClick={[Function]}
    >
      Sumar
    </button>
    <button
      onClick={[Function]}
    >
      Restar
    </button>
    <button
      onClick={[Function]}
    >
      Duplicar
    </button>
    <button
      onClick={[Function]}
    >
      Dividir
    </button>
  </div>
</Hello>
`;
